azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -135,9 +144,36 @@ protected void validateParameters() {
     }
 
     protected IFunctionAppBase<?> createOrUpdateResource(final FunctionAppConfig config) {
+        IFunctionApp app = Azure.az(AzureAppService.class).functionApp(config.resourceGroup(), config.appName());
+        final boolean newFunctionApp = !app.exists();
+        AppServiceConfig defaultConfig = !newFunctionApp ? fromAppService(app, app.plan()) : buildDefaultConfig(config.subscriptionId(),
+            config.resourceGroup(), config.appName());
+        mergeAppServiceConfig(config, defaultConfig);
+        if (!newFunctionApp && !config.disableAppInsights()) {
+            // fill ai key from existing app settings
+            config.appInsightsKey(app.entity().getAppSettings().get(CreateOrUpdateFunctionAppTask.APPINSIGHTS_INSTRUMENTATION_KEY));
+        }
         return new CreateOrUpdateFunctionAppTask(config).execute();
     }
 
+    private AppServiceConfig buildDefaultConfig(String subscriptionId, String resourceGroup, String appName) {
+        ComparableVersion javaVersionForProject = null;
+        final String outputFileName = project.getBuild().getFinalName() + "." + project.getPackaging();
+        File outputFile = new File(project.getBuild().getDirectory(), outputFileName);
+        if (outputFile.exists() && StringUtils.equalsIgnoreCase("jar", FilenameUtils.getExtension(outputFile.getName()))) {
```

Comment:
May remove the check for extension name as packaging for function should be jar

Reply:
Minor change